examples/sampling/pyg/node_classification.py

```diff
@@ -0,0 +1,184 @@
+"""
+This script demonstrates the training and testing of a GraphSAGE model for node classification on large graphs, leveraging the 
+GraphBolt framework for efficient data handling and mini-batch processing.
+
+Key Features:
+- Implements the GraphSAGE model, a scalable GNN, for node classification on large graphs.
+- Utilizes GraphBolt, an efficient framework for large-scale graph data processing.
+- Integrates with PyTorch Geometric for building and training the GraphSAGE model.
+- The script is well-documented, providing clear explanations at each step.
+
+This flowchart describes the main functional sequence of the provided example.
+main: 
+
+main
+│
+├───> Load and preprocess dataset (GraphBolt)
+│     │
+│     └───> Utilize GraphBolt's BuiltinDataset for dataset handling
+│
+├───> Instantiate the SAGE model (PyTorch Geometric)
+│     │
+│     └───> Define the GraphSAGE model architecture
+│
+├───> Train the model
+│     │
+│     ├───> Mini-Batch Processing with GraphBolt
+│     │     │
+│     │     └───> Efficient handling of mini-batches using GraphBolt's utilities
+│     │
+│     └───> Training Loop
+│           │
+│           ├───> Forward and backward passes
+│           │
+│           └───> Parameters optimization
+│
+└───> Evaluate the model
+      │
+      └───> Performance assessment on validation and test datasets
+            │
+            └───> Accuracy and other relevant metrics calculation
+
+
+"""
+
+
+
+
+import torch
+import torch.nn.functional as F
+from torch_geometric.nn import SAGEConv
+from dgl.graphbolt import BuiltinDataset, DataLoader, ItemSampler, FeatureFetcher, CopyTo
+
+
+
+
+class GraphSAGE(torch.nn.Module):
+    #####################################################################
+    # (HIGHLIGHT) Define the GraphSAGE model architecture.
+    # 
+    # - This class inherits from `torch.nn.Module`.
+    # - Two convolutional layers are created using the SAGEConv class from PyG.
+    # - 'in_size', 'hidden_size', 'out_size' are the sizes of
+    #   the input, hidden, and output features, respectively.
+    # - The forward method defines the computation performed at every call.
+    #####################################################################
+    def __init__(self, in_size, hidden_size, out_size):
+        super(GraphSAGE, self).__init__()
+        self.conv1 = SAGEConv(in_size, hidden_size)
+        self.conv2 = SAGEConv(hidden_size, out_size)
+
+    def forward(self, blocks, x, device):
+        h = x
+        for i, block in enumerate(blocks):
+            edge_index = block.edges()
+            edge_index = torch.stack([edge_index[0], edge_index[1]], dim=0).to(device)
```

Comment:
```suggestion
            edge_index = torch.stack([edge_index[0], edge_index[1]], dim=0)
```
Should already be on the correct device due to `CopyTo` in the dataloader.